### PR TITLE
Changes to show PostCall screen for attendee

### PR DESCRIPTION
**Pull request by dailam-msft**
# What
Changes to show postCall screen when "Start a Call as Attendee" is selected from home screen

# Why
Show postcall surveys for attendee customers

# How Tested
unit tested and locally
**Presenter Flow -** 
image

**Attendee Flow -** 
image

### CODE REVIEW

File: client/src/components/teams/TeamsMeetingExperience.tsx [hunk 1]
```diff
@@ -74,10 +74,8 @@ export const TeamsMeetingExperience = (props: TeamsMeetingExperienceProps): JSX.
             setRenderPostCall(true);
           });
         }
-        adapter.onStateChange((state) => {
-          if (state.call?.id !== undefined && state.call?.id !== callId) {
-            setCallId(adapter.getState().call?.id);
-          }
+        adapter.on('callIdChanged', () => {
```
[the remark this answers is not in the document]
Comment (dailam-msft) [reply client/src/components/teams/TeamsMeetingExperience.tsx]: Removed this as callID is null

File: client/src/components/rooms/RoomsMeeting.test.tsx [hunk 2]
```diff
@@ -35,6 +36,17 @@ describe('RoomsMeeting', () => {
   const roomCallLocator = {
     roomId: 'mockRoomId'
   };
```
Comment (tejaswip-msft) [inline client/src/components/rooms/RoomsMeeting.test.tsx:38]: Lets add another test saying , if postcall is present in config, then RoomsMeetingExperience is called with that in the props, else it is undefined in RoomsMeetingExperience

File: client/src/utils/PostCallUtil.ts [hunk 3]
```diff
@@ -29,3 +32,9 @@ export const submitSurveyResponseUtil = async (
     //Add Error logging here;
   }
 };
+
+export const componentToShow = (renderPostCall: boolean, userRole: RoomParticipantRole): string => {
+  if (!renderPostCall && userRole === RoomParticipantRole.presenter) return PRESENTER;
```
Comment (tejaswip-msft) [inline client/src/utils/PostCallUtil.ts:37]: lets make this simpler, 
if(!renderPostCall) {
     userRole === RoomParticipantRole.presenter ? RESENTER : ATTENDEE
}
else return SURVEY

Comment (tejaswip-msft) [reply client/src/utils/PostCallUtil.ts:37]: Also i guess we wont need this with Ralph's changes, he has 
      <CallComposite
        adapter={callAdapter}
        fluentTheme={fluentTheme}
        formFactor={formFactorValue}
        callInvitationUrl={userRole === RoomParticipantRole.presenter ? roomsInfo.inviteParticipantUrl : undefined}
      />
      so we would just need anothe check in RoomsMeetingExperience for renderPostCall and we can get rid of this function

File: client/src/components/rooms/RoomsMeetingExperience.tsx [hunk 4]
```diff
@@ -43,17 +59,49 @@ export const RoomsMeetingExperience = (props: RoomsMeetingExperienceProps): JSX.
...
+                await callAdapter.joinCall();
+                setRenderPostCall(false);
+              }}
+            />
+          );
+        break;
+      default: {
+        // this is triggered in case postCall config is not specified. It will show default postCall screen - with rejoin button
+        if (userRole === RoomParticipantRole.presenter)
+          return (
```
Comment (tejaswip-msft) [inline client/src/components/rooms/RoomsMeetingExperience.tsx:95]: we can simplify this also to 
      <CallComposite
        adapter={callAdapter}
        fluentTheme={fluentTheme}
        formFactor={formFactorValue}
        callInvitationUrl={userRole === RoomParticipantRole.presenter ? roomsInfo.inviteParticipantUrl : undefined}
      />

File: client/src/components/rooms/RoomsMeetingExperience.test.tsx [hunk 5]
```diff
@@ -82,4 +104,58 @@ describe('RoomsMeetingExperience', () => {
     expect(callComposite.first().props().adapter.getState().displayName?.includes('Virtual appointments User'));
     expect(callComposite.first().props().callInvitationUrl).toBeUndefined;
   });
+
+  it('should render Survey component when postcall is defined and valid', async () => {
```
Comment (tejaswip-msft) [inline client/src/components/rooms/RoomsMeetingExperience.test.tsx:108]: lets say, it('should render Survey component when post call is defined and user is a attendee')
and lets add one more test saying it('should not render Survey component if user is a presenter')

**Pull request closed without merging**